### DISCUSSION TITLE
是否有支持明文模式的计划?

### DISCUSSION THREAD

**Discussion by greatdiviner** [Tencent/tquic#201] [Ideas（新功能的建议和讨论）]
很多场景下, 应用并不需要quic的加密功能, 或者应用层自己会对数据做加密.

**Comment by glime-ai** [comment]
了解您的需求。在TQUIC的使用场景中，虽然QUIC协议本身包含了加密功能，但是如果您的应用层已经实现了自己的加密机制，或者在某些特定场景下不需要额外的传输层加密，您可能会考虑是否可以在不使用QUIC加密功能的情况下使用TQUIC。

根据提供的文档和讨论，TQUIC是高度可配置的，并且支持多种传输层特性，包括但不限于零RTT连接建立、多路径使用、以及高级拥塞控制算法。然而，关于直接禁用QUIC协议的加密功能的具体细节，在提供的文档中并没有直接提及。

QUIC协议的设计中，加密是其核心特性之一，用于确保传输的安全性和隐私。因此，即使应用层实现了自己的加密机制，QUIC协议层的加密通常仍然是被启用的，以保证协议本身的安全性和完整性。

如果您的应用场景确实需要在不使用QUIC加密的情况下进行通信，这可能需要对TQUIC进行更深入的定制开发，或者在应用层之上实现特定的加密/解密机制，而不是依赖于传输层的加密功能。

建议您参考TQUIC的官方文档和API参考，以了解更多关于其配置和使用的详细信息。如果官方文档中没有提供足够的信息，您也可以考虑在TQUIC的官方讨论区提出您的具体需求，以获得更专业的指导和支持。

**Reply by iyangsj** [reply]
没有正式支持的计划，有需求的用户可以按需定制
（可参考 https://datatracker.ietf.org/doc/draft-banks-quic-disable-encryption/ ）

**Reply by iyangsj** [reply]
[TQUIC 1.3.0版本](https://github.com/Tencent/tquic/releases/tag/v1.3.0)已支持 https://datatracker.ietf.org/doc/draft-banks-quic-disable-encryption/
